sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -185,7 +185,8 @@ static int skl_hda_audio_probe(struct platform_device *pdev)
 	/* Force AW88399 detection for Lenovo Legion - auto-detection may fail due to timing */
 	if (mach->mach_params.subsystem_vendor == 0x17aa &&
 	    (mach->mach_params.subsystem_device == 0x3906 ||
-	     mach->mach_params.subsystem_device == 0x3907)) {
+	     mach->mach_params.subsystem_device == 0x3907 ||
+	     mach->mach_params.subsystem_device == 0x3d6c)) {
```

Comment:
Would it be better if you have a list array and share between the 2 files? Someone might add a new ID for amp_type but forget to add it for i2s_link_mask.